src/lib.rs

```diff
@@ -145,6 +156,43 @@ impl<'data> Iterator for DataFormatDescriptorIterator<'data> {
     }
 }
 
+struct KeyValueDataIterator<'data> {
+    data: &'data [u8],
+}
+
+impl<'data> Iterator for KeyValueDataIterator<'data> {
+    type Item = (&'data [u8], &'data [u8]);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.data.len() < 4 {
+            return None;
+        }
+
+        let mut offset = 0;
+
+        let length = bytes_to_u32(self.data, &mut offset).unwrap();
+
+        let key_and_value = &self.data[offset..offset + length as usize];
+
+        // The key is terminated with a NUL character.
+        let key_end_index = key_and_value.iter().position(|&c| c == b'\0').unwrap();
```

Comment:
We should not panic on malformed inputs.
```suggestion
        let key_end_index = key_and_value.iter().position(|&c| c == b'\0')?;
```